src/alerts_utils.cc

```diff
@@ -360,45 +370,16 @@ alert_load_state (zlistx_t *alerts, const char *path, const char *filename)
     return rv;
 }
 
-/* fty_proto_zpl automatically encloses values in quotes, like this:
-
-description = "Rule changed"
-
-But JSON contains quotes on its own, and ZPL doesn't support escaping them:
-
-description = "{ "key": "Device {{var1}} does not provide expected data. It may be offline or not correctly configured.", "variables": { "var1": "sc21543331131" } }"
-
-However, ZPL doesn't require quotes and if the drop them, the value is still parsed correctly:
-
-description = { "key": "Device {{var1}} does not provide expected data. It may be offline or not correctly configured.", "variables": { "var1": "sc21543331131" } }
-
-This function drops the quotes added by fty_proto_zpl if they are enclosing JSON string.
-*/
-
-void
-s_drop_quotes_around_json (const char *state_file, const char *path, const char *filename)
+// right-pad the string with spaces to specified size
+uint8_t *
+s_pad (const char *src, size_t padded_size)
 {
-    std::ifstream input_stream (state_file);
-    char *real_state_file = zsys_sprintf ("%s/%s", path, filename);
-    std::ofstream output_stream (real_state_file);
-    std::string line;
-    while (std::getline (input_stream, line)) {
-        size_t json_start = line.find ("\"{");
-        if (json_start != std::string::npos) {
-            size_t json_end = line.rfind ("}\"");
-            if (json_end != std::string::npos) {
-                line.replace (json_start, 2, " {");
-                line.replace (json_end, 2, "} ");
-            }
-        }
-        output_stream << line << std::endl;
-    }
+    uint8_t *padded = (uint8_t *) zmalloc ((padded_size) * sizeof (uint8_t));
+    memcpy (padded, src, strlen (src));
+    for (size_t i = strlen (src); i < padded_size; i++)
```

Comment:
Would a `memset` do a better job than a loop here (potentially compiled to use some CPU instruction to optimize this operation)?

Reply:
This will copy at most three characters, so I am not sure that memset makes sense...

Reply:
ok, probably won't pay off then :)